File: app/src/main/java/net/squanchy/favorites/FavoritesModule.java
package net.squanchy.favorites;

import net.squanchy.service.firebase.FirebaseAuthService;
import net.squanchy.service.repository.EventRepository;

import dagger.Module;
import dagger.Provides;

@Module
class FavoritesModule {

    @Provides
    FavoritesService favoritesService(
            FirebaseAuthService authService,
            EventRepository eventRepository
    ) {
        return new FavoritesService(authService, eventRepository);
    }
}

File: app/src/main/java/net/squanchy/favorites/FavoritesModule.kt
package net.squanchy.favorites

import dagger.Module
import dagger.Provides
import net.squanchy.schedule.ScheduleModule
import net.squanchy.schedule.ScheduleService
import net.squanchy.service.repository.AuthService

@Module(includes = [ScheduleModule::class])
class FavoritesModule {

    @Provides
    internal fun favoritesService(
        authService: AuthService,
        scheduleService: ScheduleService
    ): FavoritesService = FirestoreFavoritesService(authService, scheduleService)
}
